controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -293,5 +293,5 @@ func (h *Helper) Patch(ctx context.Context) error {
 	log := ctrl.LoggerFrom(ctx)
 	log.V(5).Info("Patching object", "Patch", string(h.patch))
 
-	return h.client.Patch(ctx, h.original, client.RawPatch(types.MergePatchType, h.patch))
+	return h.client.Patch(ctx, h.original.DeepCopyObject().(client.Object), client.RawPatch(types.MergePatchType, h.patch))
```

Comment:
This is for preventing side effects on the object

Reply:
Side effect where?

Reply:
h.original gets modified by client.Patch

Reply:
Should we add a comment?

Reply:
Done!